back-end/app/Repositories/Implementations/Common.php

```diff
@@ -0,0 +1,93 @@
+<?php declare(strict_types=1);
+
+namespace App\Repositories\Implementations;
+
+use App\Repositories\Interfaces\Common as CommonInt;
+use Exception;
+use Illuminate\Database\Eloquent\Collection;
+use Illuminate\Database\Eloquent\Model;
+use Illuminate\Support\Facades\Log;
+
+class Common implements CommonInt
+{
+    /** @var Model $model */
+    private $model;
+
+    public function __construct(Model $model)
+    {
+        $this->model = $model;
+    }
+
+    /**
+     * Get all record from DB.
+     *
+     * @return Collection
+     */
+    public function getAll(): Collection
+    {
+        return $this->model::paginate(15);
+    }
+
+    /**
+     * Get the specified record.
+     *
+     * @param int $id
+     *
+     * @return array
+     */
+    public function getById (int $id): array
+    {
+        $record = [];
+        $model = $this->model->find($id);
+        if ($model !== null) {
+            $record = $model->getAttributes();
+            unset($record['created_at'], $record['updated_at']);
+        }
+
+        return $record;
+    }
+
+    /**
+     * @param array $attributes
+     *
+     * @return mixed
+     */
+    public function create(array $attributes)
+    {
+        $model = [];
+
+        try {
+            $model = $this->model::create($attributes);
+        } catch (Exception $e) {
+            Log::error($e->getMessage());
```

Comment:
Farei propagare l'eccezione in maniera da gestire tutto il flusso di errori con eccezioni